contracts/OperatorRegistry.sol

```diff
@@ -0,0 +1,193 @@
+
+/// @title OperatorRegistry
+/// @notice OperatorRegistry is the main contract that manages operators.
+contract OperatorRegistry is
+    IOperatorRegistry,
+    PausableUpgradeable,
+    AccessControlUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    IStakeHub public constant STAKE_HUB =
+        IStakeHub(0x0000000000000000000000000000000000002002);
+    address public override preferredDepositOperator;
+    address public override preferredWithdrawalOperator;
+
+    EnumerableSet.AddressSet private operatorSet;
+
+    // @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Initialize the OperatorRegistry contract.
+    function initialize() external initializer {
+        __AccessControl_init();
+        __Pausable_init();
+        __ReentrancyGuard_init();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    /// @notice Allows an operator that was already staked on the BNB stake manager
+    /// to join the BNBX protocol.
+    /// @param _operator Address of the operator.
+    function addOperator(
+        address _operator
+    )
+        external
+        override
+        whenNotPaused
+        nonReentrant
+        whenOperatorDoesNotExist(_operator)
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        if (_operator == address(0)) revert ZeroAddress();
+
+        (uint256 createdTime, bool jailed, ) = STAKE_HUB.getValidatorBasicInfo(
+            _operator
+        );
+        if (createdTime == 0) revert OperatorNotExisted();
+        if (jailed) revert OperatorJailed();
+
+        operatorSet.add(_operator);
+
+        emit AddedOperator(_operator);
+    }
+
+    /// @notice Allows to remove an operator from the registry.
+    /// @param _operator Address of the operator.
+    function removeOperator(
+        address _operator
+    )
+        external
+        override
+        whenNotPaused
+        nonReentrant
+        whenOperatorDoesExist(_operator)
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        if (_operator == address(0)) revert ZeroAddress();
+        if (preferredDepositOperator == _operator)
+            revert OperatorIsPreferredDeposit();
+        if (preferredWithdrawalOperator == _operator)
+            revert OperatorIsPreferredWithdrawal();
+
+        if (
+            IStakeCredit(STAKE_HUB.getValidatorCreditContract(_operator))
+                .getPooledBNB(address(this)) != 0
+        ) revert DelegationExists();
+
+        operatorSet.remove(_operator);
+
+        emit RemovedOperator(_operator);
+    }
+
+    /// @notice Allows to set the preferred operator for deposits.
+    /// @param _operator Address of the operator.
+    function setPreferredDepositOperator(
+        address _operator
+    )
+        external
+        override
+        whenNotPaused
+        nonReentrant
+        whenOperatorDoesExist(_operator)
+        onlyRole(DEFAULT_ADMIN_ROLE)
```

Comment:
onlyRole(OPERATOR)

This is a new role to control internal operations in a programmatic manner.